File: Programs/Meta/Pch.py
# Copyright mzoesch. All rights reserved.

from enum import Enum


class PchUsage(Enum):
    ALLOW = 1
    PROHIBIT = 0


def pch_usage_to_bool(pch_usage: PchUsage) -> bool:
    if pch_usage == PchUsage.ALLOW:
        return True

    if pch_usage == PchUsage.PROHIBIT:
        return False

    raise ValueError(f'Cannot convert [{pch_usage}] to bool.')


def pch_usage_to_string(pch_usage: PchUsage) -> str:
    if pch_usage == PchUsage.ALLOW:
        return 'Allow'

    if pch_usage == PchUsage.PROHIBIT:
        return 'Prohibit'

    return 'Unknown'


def volatile_pch_usage_to_string(pch_usage: PchUsage) -> str:
    if pch_usage == PchUsage.ALLOW:
        return 'Allow'

    if pch_usage == PchUsage.PROHIBIT:
        return 'Prohibit'

    raise ValueError('Unknown pch_usage in volatile_pch_usage_to_string.')

File: Programs/Core/Pch.go
// Copyright mzoesch. All rights reserved.

package Core

type PchUsage int

const (
    PCH_ALLOW    PchUsage = iota
    PCH_PROHIBIT PchUsage = iota
)

func (pch PchUsage) IsAllowed() bool {
    return pch == PCH_ALLOW
}

func (pch PchUsage) IsProhibited() bool {
    return pch == PCH_PROHIBIT
}

func (pch PchUsage) ToString() string {
    switch pch {
    case PCH_ALLOW:
        return "Allow"
    case PCH_PROHIBIT:
        return "Prohibit"
    default:
        panic("Unknown PchUsage value.")
    }
}

func PchUsageFromString(str string) PchUsage {
    switch str {
    case "Allow", "ALLOW":
        return PCH_ALLOW
    case "Prohibit", "PROHIBIT":
        return PCH_PROHIBIT
    default:
        panic("Unknown PchUsage value.")
    }
}
